app/serializers/topic_view_details_serializer.rb

```diff
@@ -144,6 +145,10 @@ def include_can_archive_topic?
     scope.can_archive_topic?(object.topic)
   end
 
+  def include_can_edit_staff_notes?
+    scope.can_edit_staff_notes?(object.topic)
```

Comment:
Since `can_edit_staff_notes` is just an alias of `can_close_topic` and `can_archive_topic`, we should memoize the results so that in the worst case we don't have to run `GroupUser.where(group_id: topic.category.reviewable_by_group_id, user_id: @user.id).exists?` 3 times.

Reply:
Agreed, memoization here is a good idea!